### PR TITLE
GuniProxyActions wrapper contract implementation

### PR DESCRIPTION
**Pull request by fenris85**
GUniProxyActions implementation contains two main public methods: 
- openMultiplyGuniVault  - allows to open new vault, take flashloan, swap part of DAI to USDC wrap into Guni token and open a vault with it.
- closeGuniVaultExitDai - oposite operation to payback vault debt with a flashloan, exit Guni collateral, unwrap it, swap USCD to DAI and exit resting amount of DAI.

### CODE REVIEW

File: tenderly.yaml [hunk 1]
```diff
@@ -13,6 +13,6 @@ exports:
       petersburg_block: 0
       berlin_block: 0
     forked_network: mainnet
-    project_slug: jinx/project
+    project_slug: fenris85tend/project
```
Comment (peculiarity) [inline tenderly.yaml:16]: Don't commit these

Comment (fenris85) [reply tenderly.yaml:16]: yeah, I noticed this after comiting. I plan to move to non-WIP branch everything shortly, so I'll remove it from here

File: contracts/multiply/Guni.sol [hunk 2]
```diff
@@ -0,0 +1,659 @@
...
+  function onFlashLoan(
+    address initiator,
+    address token,
+    uint256 amount,
+    uint256 fee,
+    bytes calldata data
+  ) external returns (bytes32);
+}
+
+interface IERC3156FlashLender {
```
Comment (peculiarity) [inline contracts/multiply/Guni.sol:166]: I don't se this being used anywere. Also can't you use the existing interface definition ?

File: contracts/multiply/Guni.sol [hunk 3]
```diff
@@ -0,0 +1,659 @@
...
+      addressRegistry.multiplyProxyActions,
+      1
+    );
+
+    IManager(addressRegistry.manager).cdpAllow(
+      cdpData.cdpId,
+      guniAddressRegistry.guniProxyActions,
+      1
+    );
+    ILendingPoolV2 lendingPool = getAaveLendingPool(addressRegistry.aaveLendingPoolProvider);
```
Comment (peculiarity) [inline contracts/multiply/Guni.sol:303]: WHy do you use aave?

File: contracts/multiply/Guni.sol [hunk 4]
```diff
@@ -0,0 +1,659 @@
...
+
+    uint256 guniBalance;
+    uint256 bal1 = otherToken.balanceOf(address(this));
+
+    {
+      GUNIRouterLike router = GUNIRouterLike(guniAddressRegistry.router);
+      dai.approve(address(router), bal0);
+      otherToken.approve(address(router), bal1);
+
+      (, , guniBalance) = router.addLiquidity(address(guni), bal0, bal1, 0, 0, address(this));
```
Comment (adamskrodzki) [inline contracts/multiply/Guni.sol:399]: 0,0, opens it to front running attacks

File: contracts/multiply/Guni.sol [hunk 5]
```diff
@@ -0,0 +1,659 @@
...
+    );
+
+    GUNITokenLike guni = GUNITokenLike(guniAddressRegistry.guni);
+
+    uint256 guniBalance = guni.balanceOf(address(this));
+
+    {
+      GUNIRouterLike router = GUNIRouterLike(guniAddressRegistry.router);
+      guni.approve(address(router), guniBalance);
+      router.removeLiquidity(address(guni), guniBalance, 0, 0, address(this));
```
Comment (adamskrodzki) [inline contracts/multiply/Guni.sol:449]: 0, 0, opens front running attacks

**Pull request closed without merging**